### ISSUE TITLE
Incapacitation keyboard lockout has a critical side effect... Zeus can not release AI while incapacitated.

### ISSUE THREAD

**Issue by Sycholic** [acemod/ACE3#6354]
**Arma 3 Version:**  1.82  stable
**CBA Version:** 3.7.0
**ACE3 Version:** current to Steam WS

**Mods:**
```
- CBA_A3
- ace
```

**Description:**
- if you enable AI 'incapacitation' aka no instant death for AI.  You can not play as zeus ever and control AI.  If you do... you end up being stuck 'incapacitated' as the AI and cant not return to zeus menu.  you either have to be revived, bleed out, log out to the lobby, or hope someone executes you (if thats even enabled also..)

Apparently this has been a long time issue with ACE.  why no-one has never once reported or asked for the Zeus hotkey (trigger action whatever..) to be 'unblocked' like how the ESC key is not blocked.

**Steps to reproduce:**
- Enable AI incapacitation, take over an AI as zeus then get shot and incapacitated while remote controlling that AI.

**Where did the issue occur?**
- In Game, MP/SP

**Comment by Sycholic** [comment]
This also is a issue for captives as well.  If you are as Zeus and get captured you can not release the AI either.  The Zeus menu needs to be accessible regardless of the state an AI is in.

**Comment by commy2** [comment]
Needs IsNotCaptive or whatever it was called exception to the zeus interaction menu actions.

**Comment by Sycholic** [comment]
thinking its a logic issue in line 74 of  ACE3/addons/common/functions/fnc_disableUserInput.sqf  and it needs to check to getAssignedCuratorUnit?  because if you are controlling an AI Im suspecting that the logic check of {getAssignedCuratorLogic player in allCurators} does not ring true when a zeus is remote controlling.  Im working on setting up a workspace to do some code testing.  Its not about the zeus interaction menu, its the incapacitation effect ends up blocking almost all forms of inputs

**Comment by dedmen** [comment]
@Sycholic you can link btw
https://github.com/acemod/ACE3/blob/ffaa195fe576625f052eeee04b1af9f3f377434f/addons/common/functions/fnc_disableUserInput.sqf#L74-L77

>Im suspecting that the logic check of {getAssignedCuratorLogic player in allCurators} does not ring true when a zeus is remote controlling

When remote controlling `player` is still the zeus unit and not the remote controlled one so that shouldn't be an issue.

**Comment by BaerMitUmlaut** [comment]
Are you using a custom keybind for Zeus, and if so, which one?

**Comment by SilentSpike** [comment]
[Here's how I handle zeus in spectator for reference](https://github.com/acemod/ACE3/blob/104efe28bcb0ef9661d5e1b9bd860945c260550b/addons/spectator/functions/fnc_ui_handleKeyDown.sqf#L181)

**Comment by SilentSpike** [comment]
Also, not part of this issue, but [the escape menu can also be handled much more elegantly too](https://github.com/acemod/ACE3/blob/104efe28bcb0ef9661d5e1b9bd860945c260550b/addons/spectator/functions/fnc_ui_handleKeyDown.sqf#L39).

**Comment by Sycholic** [comment]
Not sure then the issue is there.  This is the core reason why staff doesnt allow AI incapacitation.  My thought was that the zeus check is failing because you arent in the zeus menu, you are controlling an AI.    We use zeus as both 'virtual' and 'players' admin staff can use zeus as a player but zeus staff only uses the virtual. (not sure if that is relative could be)

**Comment by BaerMitUmlaut** [comment]
The Ctrl might be the issue, can you try setting it to a single key press?

**Comment by dedmen** [comment]
@SilentSpike https://github.com/acemod/ACE3/blob/104efe28bcb0ef9661d5e1b9bd860945c260550b/addons/spectator/functions/fnc_ui_handleKeyDown.sqf#L181
actionKeys goes bonkers very quickly if the key has modifiers aka ctrl/shift/alt

**Comment by Sycholic** [comment]
okay update after doing some testing. turns out its not an issue with the Zeus key but when you are a player zeus (vs a virtual zeus).  the 'release UAV' scrollwheel option is not available... wasnt a issue with the zeus hotkey.  My problem was my own fault but after testing found out the other person having the issue was referring to the scrollwheel option release as that is how his primary method of releasing AI was and wasnt using the Zeus key to jump back.   Probably can call this closed, or keep it open and mark it as possible todo/feature.  and unblock the 'release UAV' scrollwheel cmd available to those with zeus ability?

**Comment by commy2** [comment]
>when you are a player zeus (vs a virtual zeus). the 'release UAV' scrollwheel option is not available

What is a "player zeus" and how would one ever abort remote control if no user action is available for that?

**Comment by commy2** [comment]
>actionKeys goes bonkers very quickly if the key has modifiers aka ctrl/shift/alt

Yes, `_key in (actionKeys "CuratorInterface")` is broken, because every usage of `actionKeys` is broken.

**Comment by bux** [comment]
> player zeus

You're controlling a unit that you've selected in the lobby that has an assigned curator logic

**Comment by Sycholic** [comment]
Player Zeus' aka a player that has picked a player side (bluefor in this case) not 'virtual' from the lobby.  Only Admins have that ability all other zeus's use the Virtual login from the lobby. (we dont have voting zeus its a whitelist/staff only)

**Comment by SilentSpike** [comment]
There's a lot of terminology being thrown around here which is confusing things because it has specific meanings when it comes to scripting which differ from colloquial use.

As I understand it, the actual issue here is that when unconscious, the display which is open prevents zeus from leaving the remote controlled unit. This seems to be because `actionKeys` doesn't work with modifier keys and also because the display blocks the vanilla scroll wheel action menu (which there's nothing we can really do to fix).

**Comment by SilentSpike** [comment]
Simple solution on the user end is to bind the zeus key to a single key without modifier and use that.

On development end, I'm not really sure what we can do. It's possible we should kick zeus out of a remote controlled unit when it goes unconscious, but that would also be annoying if it was just temporary.

**Comment by Sycholic** [comment]
if there nothing you can do about the scrollwheel then there nothing that can be done if its a vanilla blocking.  The keybind issue I had was my own fault I figured out when we were just testing last night.  I'd say consider it closed.  Thank you all for the feedback good to see yas active.

**Comment by dedmen** [comment]
The actual issue was the user trying to exit remote control via the scroll menu. Which you can't use when unconcious. I actually didn't even know that you could use the scroll menu for that. I always just use the Zeus button to stop remote controlling.

In the end it was just user error. Not ACE error. The problem with the actionKey not supporting modifiers theoretically still stands. But no one reported that being a problem yet so I guess there are not really people who rebind the zeus key to something with a modifier.

**Comment by Sycholic** [comment]
yeah my problem was separate from the admin level zeus's issue. my own was user operator error... just not paying attention and ended up undoing keybinds i had set up and the issue for him was resolved same time discussing this out, and thats BI's problem so thank you everyone who replied and helped out.
